pkg/controller/association/conf.go

```diff
@@ -125,14 +125,15 @@ func AllowVersion(resourceVersion version.Version, associated commonv1.Associate
 		}
 		refVer, err := version.Parse(assoc.AssociationConf().Version)
 		if err != nil {
-			logger.Error(err, "Invalid version found in association conf", "association_version", refVer)
+			logger.Error(err, "Invalid version found in association conf", "association_version", assoc.AssociationConf().Version)
 			return false
 		}
 		if !refVer.IsSameOrAfter(resourceVersion) {
 			// the version of the referenced resource (example: Elasticsearch) is lower than
 			// the desired version of the reconciled resource (example: Kibana)
-			logger.Info("Delaying version deployment since an associated resource is not upgraded yet",
-				"version", resourceVersion, "ref_namespace", assocRef.Namespace, "ref_name", assocRef.Name, "ref_version", refVer)
+			msg := "Delaying version deployment since an associated resource is not upgraded yet"
+			logger.Info(msg, "version", resourceVersion, "ref_namespace", assocRef.Namespace, "ref_name", assocRef.Name, "ref_version", refVer)
+			recorder.Event(associated, corev1.EventTypeWarning, events.EventReasonDelayed, msg)
```

Comment:
Isn't this lacking details? I'm not sure if users would immediately jump from `associated resource` to looking at `elasticsearchRef` or `kibanaRef`.

Reply:
👍 changing the event msg to `fmt.Sprintf("Delaying version deployment since the referenced %s is not upgraded yet", assoc.AssociatedType()))`.